salt/metalk8s/kubernetes/etcd/installed.sls

```diff
@@ -4,19 +4,33 @@ include:
   - metalk8s.kubernetes.ca.etcd.advertised
   - .certs
 
-{%- set host_name = grains['id'] %}
-{%- set host = grains['metalk8s']['control_plane_ip'] %}
+{%- set node_name = grains['id'] %}
+{%- set node_ip = grains['metalk8s']['control_plane_ip'] %}
 
-{%- set endpoint  = host_name ~ '=https://' ~ host ~ ':2380' %}
+{%- set endpoint = 'https://' ~ node_ip ~ ':2380' %}
 
-{#- Get the list of existing etcd node. #}
-{%- set etcd_endpoints = pillar.metalk8s.etcd.members %}
+{#- Get the list of existing etcd member. #}
+{%- set etcd_members = pillar.metalk8s.etcd.members %}
 
 {#- Compute the initial state according to the existing list of node. #}
-{%- set state = "existing" if etcd_endpoints else "new" %}
+{%- set state = "existing" if etcd_members else "new" %}
```

Comment:
Not sure this is the right approach (though indeed, it's not related to this PR/issue).

In essence, what we're saying here is "If all nodes are unavailable (which causes the member list to be empty), we consider this to be a new cluster", which is definitely not a given. Actually, only during very first bootstrap this can be the case, otherwise we're disaster-recovering or so.

Worth an issue IMO.

Reply:
I agree it's not a safe approach
#2102